src/main/java/org/shanerx/tradeshop/objects/IllegalItemList.java

```diff
@@ -0,0 +1,146 @@
+        this.list = list;
+    }
+
+    public ListType getType() {
+        return type;
+    }
+
+    public void setType(ListType newType) {
+        this.type = newType;
+    }
+
+    public void setType(String newType) {
+        setType(ListType.valueOf(newType));
+    }
+
+    public boolean isIllegal(Material mat) {
+        switch (type) {
+            case WHITELIST:
+                return !list.contains(mat);
+            case BLACKLIST:
+                return list.contains(mat);
+            case DISABLED:
+            default:
+                return false;
+        }
+    }
+
+    public void clear() {
+        list.clear();
+    }
+
+    public boolean add(Material mat) {
+        return list.add(mat);
+    }
+
+    public boolean remove(Material mat) {
+        return list.remove(mat);
+    }
+
+    public void addColourSet(String mat) {
+        List<String> colourSet = Arrays.asList("BLACK", "RED", "GREEN", "BROWN", "BLUE", "PURPLE", "LIGHT_BLUE", "LIGHT_GRAY", "GRAY", "PINK", "LIME", "YELLOW", "CYAN", "MAGENTA", "ORANGE", "WHITE");
+
+        add(Material.matchMaterial(mat));
+        for (String colour : colourSet) {
+            add(Material.matchMaterial(colour + "_" + mat));
+        }
+
+    }
+
+    public void checkAndAdd(String check) {
+        Material mat = Material.matchMaterial(check);
+        if (mat != null) {
+            add(mat);
+        } else {
+            switch (check.toLowerCase()) {
+                case "@banner":
+                    addColourSet("BANNER");
+                    addColourSet("WALL_BANNER");
+                    break;
+                case "@candle":
+                    addColourSet("CANDLE");
+                    break;
+                case "@carpet":
+                    add(Material.MOSS_CARPET);
+                    addColourSet("CARPET");
+                    break;
+                case "@concrete":
+                    addColourSet("CONCRETE");
+                    break;
+                case "@concrete_powder":
+                    addColourSet("CONCRETE_POWDER");
+                    break;
+                case "@dye":
+                    addColourSet("DYE");
+                    break;
+                case "@glazed_terracotta":
+                    addColourSet("GLAZED_TERRACOTTA");
+                    break;
+                case "@shulker_box":
+                    addColourSet("SHULKER_BOX");
+                    break;
+                case "@stained_glass":
+                    addColourSet("STAINED_GLASS");
+                    break;
+                case "@stained_glass_pane":
+                    addColourSet("STAINED_GLASS_PANE");
+                    break;
+                case "@terracotta":
+                    addColourSet("TERRACOTTA");
+                    break;
+                case "@wool":
+                    addColourSet("WOOL");
+                    break;
+            }
+        }
+    }
+}
```

Comment:
Has this system been tested?

Reply:
I did basic testing on it and had no problems with any lists/moses